ownCloud/ownCloud/Bookmarks/Connection Issues /IssuesViewController.swift

```diff
@@ -0,0 +1,217 @@
+        issuesTableView.bounces = false
+        issuesTableView.showsVerticalScrollIndicator = false
+
+        self.view.addSubview(issuesTableView)
+
+        issuesTableView.bottomAnchor.constraint(equalTo: bottomContainer.topAnchor, constant: -20).isActive = true
+        issuesTableView.leftAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.leftAnchor, constant: 20).isActive = true
+        issuesTableView.rightAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.rightAnchor, constant: -20).isActive = true
+        issuesTableView.topAnchor.constraint(greaterThanOrEqualTo: self.view.safeAreaLayoutGuide.topAnchor, constant: 20).isActive = true
+        tableHeighConstraint = issuesTableView.heightAnchor.constraint(equalToConstant: issuesTableView.contentSize.height)
+        tableHeighConstraint?.isActive = true
+
+        issuesTableView.tableHeaderView?.backgroundColor = UIColor.white
+    }
+
+    override func viewWillLayoutSubviews() {
+        super.updateViewConstraints()
+        self.tableHeighConstraint?.constant =  self.issuesTableView.contentSize.height
+    }
+
+    func addBottomContainerElements(elements: [UIView]) {
+        for element in elements {
+            self.bottomContainer.addSubview(element)
+        }
+    }
+
+    @objc func dismissView() {
+        self.dismiss(animated: true, completion: nil)
+    }
+}
+
+extension IssuesViewController: UITableViewDelegate {
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return UITableViewAutomaticDimension
+    }
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+
+        guard self.headerTitle != nil else {
+            return 0
+        }
+
+        return 40
+    }
+}
+
+extension IssuesViewController: UITableViewDataSource {
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let cell = UITableViewCell(style: .default, reuseIdentifier: nil)
+        cell.backgroundColor = .white
+        cell.textLabel?.attributedText = NSAttributedString(string: self.headerTitle!, attributes: [.font : UIFont.systemFont(ofSize: 20, weight: .semibold)])
+
+        let separatorView: UIView = UIView()
+        separatorView.translatesAutoresizingMaskIntoConstraints = false
+        separatorView.backgroundColor = UIColor(hex: 0x04040F)
+        cell.addSubview(separatorView)
+        separatorView.heightAnchor.constraint(equalToConstant: 0.25).isActive = true
+        separatorView.leftAnchor.constraint(equalTo: cell.leftAnchor, constant: 0).isActive = true
+        separatorView.rightAnchor.constraint(equalTo: cell.rightAnchor, constant: 0).isActive = true
+        separatorView.bottomAnchor.constraint(equalTo: cell.bottomAnchor, constant: 0).isActive = true
+
+        cell.layoutSubviews()
```

Comment:
From the [Apple docs](https://developer.apple.com/documentation/uikit/uiview/1622482-layoutsubviews):

> You should not call this method directly. If you want to force a layout update, call the setNeedsLayout() method instead to do so prior to the next drawing update. If you want to update the layout of your views immediately, call the layoutIfNeeded() method.

Please avoid calling layoutSubviews() and call setNeedsLayout() and layoutIfNeeded() if absolutely needed.

In this case, however, I'm not sure you really need to call any of these. IIRC adding constraints should already trigger layout.